src/arc-drop.md

```diff
@@ -0,0 +1,99 @@
+# Dropping
+
+We now need a way to decrease the reference count and drop the data once it is
+low enough, otherwise the data will live forever on the heap.
+
+To do this, we can implement `Drop`.
+
+Basically, we need to:
+1. Decrement the reference count
+2. If there is only one reference remaining to the data, then:
+3. Atomically fence the data to prevent reordering of the use and deletion of
+   the data
+4. Drop the inner data 
+
+First, we'll need to get access to the `ArcInner`:
+```rust,ignore
+let inner = unsafe { self.ptr.as_ref() };
+```
+
+Now, we need to decrement the reference count. To streamline our code, we can
+also return if the returned value from `fetch_sub` (the value of the reference
+count before decrementing it) is not equal to `1` (which happens when we are not
+the last reference to the data).
+```rust,ignore
+if inner.rc.fetch_sub(1, Ordering::???) != 1 {
+    return;
+}
+```
+
+TODO
```

Comment:
Drop this TODO and remove the ??? above for now -- let's land this as a good starting point we can improve later.

Reply:
I was mostly waiting to implement your suggestions on this but got killed by the wall of text lol